internal/storage/v2/memory/tenant.go

```diff
@@ -16,18 +16,23 @@ import (
 // Tenant is an in-memory store of traces for a single tenant
 type Tenant struct {
 	sync.RWMutex
-	ids        []pcommon.TraceID // ring buffer used to evict oldest traces
-	traces     map[pcommon.TraceID]ptrace.Traces
+	ids        map[pcommon.TraceID]int
+	traces     []*tracesById // ring buffer used to store traces
```

Comment:
this does not need to be a pointer
```suggestion
	traces     []tracesById // ring buffer used to store traces
```